File: java/src/main/java/money_problem/domain/DomainExtensions.java
package money_problem.domain;

import lombok.experimental.UtilityClass;

@UtilityClass
public class DomainExtensions {
    public static Money dollars(double amount) {
        return new Money(amount, Currency.USD);
    }

    public static Money euros(double amount) {
        return new Money(amount, Currency.EUR);
    }

    public static Money koreanWons(double amount) {
        return new Money(amount, Currency.KRW);
    }

    public static Portfolio portfolioFrom(Money money1, Money money2) {
        return new Portfolio(money1, money2);
    }
}

File: kotlin/src/main/kotlin/money_problem/domain/DomainExtensions.kt
package money_problem.domain

fun Double.dollars(): Money = Money(this, Currency.USD)
fun Double.euros(): Money = Money(this, Currency.EUR)
fun Double.koreanWons(): Money = Money(this, Currency.KRW)
fun Money.addToPortfolio(money: Money): Portfolio = Portfolio(listOf(this, money))
fun Portfolio.add(money: Money): Portfolio = this.copy(moneys = this.moneys.plus(money))